packages/datasource-sql/src/types.ts

```diff
@@ -30,10 +30,18 @@ export type ProxyOptions = {
   command?: 'connect';
 };
 
+export type SshOptions = {
+  port: number;
+  host: string;
+  username: string;
+  privateKey: Buffer;
+};
+
```

Comment:
If you do the following, you get automatic documentation + allow users to use any alternative way to connect to the ssh server (password, ...)

(I'm omitting `sock` because you override it)

```suggestion
import { ConnectConfig } from 'ssh2';

export type SshOptions = Omit<ConnectConfig, 'sock'>;
```